feat(preset): dont split params between quotes

elbandi opened the pull request:
<!-- If this is your first pull request: sign the CLA with this GitHub app: https://cla-assistant.io/renovatebot/renovate -->
<!-- Make sure the `Allow edits and access to secrets by maintainers` checkbox is checked on this pull request. -->
<!-- Please read https://github.com/renovatebot/renovate/blob/main/.github/contributing.md before you create your pull request.-->

## Changes

Don't split preset params inside quote.

The split regexp come from here: https://stackoverflow.com/questions/73121226/javascript-regex-split-a-string-by-commas-but-ignore-commas-within-double-quote
And the remove quote replace from here: https://stackoverflow.com/questions/26156292/trim-specific-character-from-a-string

## Context

we have preset args with comma:

    "defaultRegistryUrlTemplate": "https://www.somesite.com/query?code={{arg0}}"

We call this:  `mypreset("foo,bar")` so the defaultRegistryUrlTemplate will be: "https://www.somesite.com/query?code=foo,bar"

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

<!-- Do you have any suggestions about this PR template? Edit it here: https://github.com/renovatebot/renovate/edit/main/.github/pull_request_template.md -->

<!-- Please do not force push to your PR's branch after you have created your PR, as doing so forces us to review the whole PR again. This makes it harder for us to review your work because we don't know what has changed. -->
<!-- PRs will always be squashed by us when we merge your work. Commit as many times as you need in this branch. -->

elbandi committed:
feat(preset): dont split params between quotes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/renovatebot/renovate?pullRequest=35347) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/renovatebot/renovate?pullRequest=35347) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/renovatebot/renovate?pullRequest=35347) it.</sub>

viceice commented on lib/config/presets/parse.ts:
```diff
@@ -45,7 +45,8 @@ export function parsePreset(input: string): ParsedPreset {
     params = str
       .slice(str.indexOf('(') + 1, -1)
-      .split(',')
+      .split(/,\s*(?=(?:(?:[^"]*"){2})*[^"]*$)/)
```
this can cause harm because of backtracking, so a no-go.

viceice replied:
@zharinov Idea? Would like to avoid lookahead /loockbehind because of possible backtracking. This can cause harm because it's user input.

Elbandi commented:
Another idea: use a different character for no-split commas, split the params, and replace specila char to comma (and no need 
 quote):
preset(foo,bar§baz) -> `.split().map(v=> v.replace("§",","))` -> ["foo", "bar,baz"] and the special char was the '§'.

Second idea: escape the comma, but still need a special char to avoid regexp lookahead /loockbehind:
preset(foo,bar\\,baz) -> `replace("\\,", "§").split().map(v=> v.replace("§",","))` ->["foo", "bar,baz"]

Or use a better regexp! (my knowledge is less for that... )
